src/core/thread/key_manager.hpp

```diff
@@ -152,6 +153,48 @@ class NetworkKey : public otNetworkKey, public Equatable<NetworkKey>
      */
     Error GenerateRandom(void) { return Random::Crypto::FillBuffer(m8, sizeof(m8)); }
 #endif
+
+} OT_TOOL_PACKED_END;
+
+/**
+ * Provides a representation for Network Key reference.
+ *
+ */
+typedef otNetworkKeyRef NetworkKeyRef;
+
+/**
+ * This class represents a Thread Network Key material.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class NetworkKeyInfo
+{
+public:
+    /**
+     * Type of Crypto used by the platform. This defines if the key is stored as a literal string, or as
+     * a reference.
+     */
+    Mac::CryptoType mCryptoType;
```

Comment:
Do we need to save `mCryptoType` for `NetworkKey`/`Pskc`?

Reading the code I think it is always tied to `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` config. 
- For example, I think independent of how the network key is set (using `SetNetworkKey()` or `SetNetworkKeyRef()` in `KeyManager` we should save in PSA storage when PSA feature is enabled

Reply:
I had the same question. I kept it there, to assist applications in runtime to determine if PSA is being used to store PSKc and NetworkKey.

Reply:
I will remove `CryptoType`.